### ISSUE TITLE
STYLE004: Detect column ordinals in ORDER BY / GROUP BY

### ISSUE THREAD

**Issue by RAprogramm** [RAprogramm/sql-query-analyzer#20]
## Summary

Add a rule to detect numeric ordinal references in `ORDER BY` and `GROUP BY` clauses (e.g., `ORDER BY 1, 2`) instead of explicit column names.

## Why This Matters

Using ordinal positions is a maintenance hazard:

- **Fragile code** — changing SELECT columns breaks ordering silently
- **Unreadable** — `ORDER BY 1, 3, 2` tells you nothing about intent
- **Refactoring nightmare** — adding/removing columns shifts numbers
- **Bug source** — off-by-one errors common when modifying
- **Code review difficulty** — reviewers can't verify correctness without counting

### The Fragility Problem

```sql
-- Original query (sorts by name, date):
SELECT id, name, created_at FROM users ORDER BY 2, 3;

-- Someone adds email to SELECT:
SELECT id, email, name, created_at FROM users ORDER BY 2, 3;

-- Now sorts by email, name — NOT what was intended!
-- No error, no warning, just wrong results.
```

### Explicit Names Are Self-Documenting

```sql
-- Bad: What is this sorting by?
SELECT id, name, created_at, status FROM users ORDER BY 2, 3 DESC, 4;

-- Good: Crystal clear intent
SELECT id, name, created_at, status FROM users ORDER BY name, created_at DESC, status;
```

## Rule Specification

| Property | Value |
|----------|-------|
| Rule ID | `STYLE004` |
| Name | Ordinal in ORDER BY/GROUP BY |
| Category | Style |
| Severity | Info |
| Default | Enabled |

### Detection Patterns

```sql
-- Should trigger
SELECT name, COUNT(*) FROM users GROUP BY 1;
SELECT * FROM orders ORDER BY 1, 2;
SELECT id, name FROM users ORDER BY 1 DESC;

-- Should NOT trigger (explicit columns)
SELECT name, COUNT(*) FROM users GROUP BY name;
SELECT * FROM orders ORDER BY created_at, id;
SELECT id, name FROM users ORDER BY id DESC;
```

### Suggested Message

> "Using column ordinal (ORDER BY 1) instead of column name. Ordinals are fragile — changing SELECT columns silently breaks ordering. Use explicit column names: ORDER BY column_name."

## Historical Context

Ordinal syntax was added to SQL for brevity in interactive sessions, not for application code. Most modern linters flag this:

- **SQLFluff**: [L054 - Group By Ordinal](https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.rules.sphinx.Rule_L054)
- **pgFormatter**: warns on ordinal usage
- **SonarQube SQL**: flags as code smell

## References

- [SQL Style Guide - Simon Holywell](https://www.sqlstyle.guide/#sorting-and-grouping) — explicitly recommends against ordinals
- [PostgreSQL ORDER BY](https://www.postgresql.org/docs/current/queries-order.html)
- [MySQL GROUP BY Handling](https://dev.mysql.com/doc/refman/8.0/en/group-by-handling.html)
- [SQLFluff Rule L054](https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.rules.sphinx.Rule_L054)

## Implementation Notes

- Detect: `ORDER BY` or `GROUP BY` followed by numeric literal
- Match patterns: `ORDER BY 1`, `ORDER BY 1, 2`, `GROUP BY 1`
- Regex-ish: `(ORDER|GROUP)\s+BY\s+\d`
- Don't flag: `LIMIT 1`, `OFFSET 10` (different context)
- Consider: `ORDER BY 1 + 0` edge case (rare, but expression vs ordinal)
